How to load extensions not in the Marketplace (#15573)
docker/docker.github.io: desktop/extensions.md

1. From the menu bar, select the ellipsis to the right of **Extensions**.
2. Click **Manage Extensions**. This displays a list of extensions you've installed.
3. Click **Uninstall**.

> ## Install an extension which is not in the Marketplace

Reviewer: ```suggestion
## Install an extension not available in the Marketplace
```

> The Extensions Marketplace is the trusted and official place to install extensions from within Docker Desktop. These extensions have gone through a review process and thus meet a minimum set of requirements. However, the Marketplace it is not the only place where to install extensions from.

Reviewer: ```suggestion
The Extensions Marketplace is the trusted and official place to install extensions from within Docker Desktop. These extensions have gone through a review process by Docker. However, other extensions can also be installed in Docker Desktop if you trust the extension author.
```

> Given the nature of a Docker Extension (i.e. a Docker image) you could find other places where users have their extensions' source code published: GitHub, GitLab or even hosted in image registries like DockerHub or GHCR.

Reviewer 2: ```suggestion
Given the nature of a Docker Extension (i.e. a Docker image) you can find other places where users have their extension's source code published. For example on GitHub, GitLab or even hosted in image registries like DockerHub or GHCR.
```

> You can install Docker Extensions that have been developed by the community or internally at your company from a teammate: you are not limited to install extensions from the Marketplace only.

Reviewer 2: ```suggestion
You can install an extension that has been developed by the community or internally at your company from a teammate. You are not limited to installing extensions just from the Marketplace.
```

> **Note**
>

> > Ensure the option "Allow only extensions distributed through the Docker Marketplace" is disabled. Otherwise this will This will prevent to install any other extension via the Extension SDK tools.

Reviewer: ```suggestion
> Ensure the option "Allow only extensions distributed through the Docker Marketplace" is disabled. Otherwise this will This will prevent to install any extension not listed in the Marketplace via the Extension SDK tools.
```

Author: That's a very good clarification, thanks!

> > Ensure the option "Allow only extensions distributed through the Docker Marketplace" is disabled. Otherwise this will This will prevent to install any extension not listed in the Marketplace via the Extension SDK tools.

Reviewer 2: ```suggestion
> Ensure the option **Allow only extensions distributed through the Docker Marketplace** is disabled. Otherwise, this prevents any extension not listed in the Marketplace, via the Extension SDK tools from, being installed.
```

> You can change this option in the Settings > Extensions.

> > **Warning**

Reviewer 2: I'd move this warning to just under the header. Also if you add {: .warning} after the note, it changes the color to red/orange. More details here https://docs.docker.com/contribute/components/call-outs/

Author: Just moved it, thanks! 😄

>

> > Docker Extensions that are not in the Marketplace haven't been gone through a review process by Docker.

Reviewer 2: ```suggestion
> Docker Extensions that are not in the Marketplace haven't gone through Docker's review process.
```

> Extensions can install binaries, invoke commands and access files on your machine. Installing them is at your own risk.

> To install an extension which is not present in the Marketplace, you can use the Extensions CLI that is bundled as part of Docker Desktop.

Reviewer: ```suggestion
To install an extension which is not present in the Marketplace, you can use the Extensions CLI that is bundled with Docker Desktop.
```

> In a terminal, type `docker extension install IMAGE[:TAG]` to install an extension by its image reference and optionally a tag. Use the `-f` or `--force` flag to not prompt for confirmation.

Reviewer: ```suggestion
In a terminal, type `docker extension install IMAGE[:TAG]` to install an extension by its image reference and optionally a tag. Use the `-f` or `--force` flag to avoid interactive confirmation.
```

Go to the Docker Dashboard to see the new extension installed.

## List installed extensions

Regardless whether the extension was installed from the Marketplace or manually by using the Extensions CLI, you can use the `docker extension ls` command to display the list of extensions installed.

> As part of the output you could see the extension ID, the provider, version, the title and whether it runs a backend container or has deployed binaries to the host, for example:

Reviewer 2: ```suggestion
As part of the output you'll see the extension ID, the provider, version, the title and whether it runs a backend container or has deployed binaries to the host, for example:
```

```
$ docker extension ls
ID                                       PROVIDER            VERSION             UI                              VM                  HOST

> docker/volumes-backup-extension          Docker Inc.         1.0.0               1 tab(Volumes Backup & Share)   Running(1)          1 binarie(s)

Reviewer: Maybe show an example of an extension that is not in the marketplace ?

Reviewer 2: +1

Author: Yes, that makes more sense, thanks!

```

> Go to the Docker Dashboard, click on "Add Extensions" and on the "Installed" tab to see the new extension installed.

Reviewer 2: ```suggestion
Go to the Docker Dashboard, click on **Add Extensions** and on the **Installed** tab to see the new extension installed.
```

> Notice that it will display an `UNPUBLISHED` label indicating that the extension has not been installed from the Marketplace.

Reviewer 2: ```suggestion
Notice that an `UNPUBLISHED` label displays which indicates that the extension has not been installed from the Marketplace.
```

## Update an extension which is not in the Marketplace

To update an extension which is not present in the Marketplace, in a terminal type `docker extension update IMAGE[:TAG]` where the `TAG` should be different from the extension that is already installed.

> 

Reviewer: Notice that after installing extensions that are not published in the Marketplace, you will not get update notifications from Docker Desktop

Author: Added a note about this

For instance, if you installed an extension with `docker extension install john/my-extension:0.0.1`, you can update it by running `docker extension update john/my-extension:0.0.2`.
Go to the Docker Dashboard to see the new extension updated.

## Uninstall an extension which is not in the Marketplace

> To uninstall an extension which is not present in the Marketplace, you could either navigate to the "Installed" tab in the Marketplace and press on the "Uninstall" button, or from a terminal type `docker extension rm IMAGE[:TAG]`.

Reviewer: ```suggestion
To uninstall an extension which is not present in the Marketplace, you could either navigate to the "Installed" tab in the Marketplace and click the "Uninstall" button, or from a terminal type `docker extension uninstall IMAGE[:TAG]`.
```

> To uninstall an extension which is not present in the Marketplace, you could either navigate to the "Installed" tab in the Marketplace and click the "Uninstall" button, or from a terminal type `docker extension uninstall IMAGE[:TAG]`.

Reviewer 2: ```suggestion
To uninstall an extension which is not present in the Marketplace, you can either navigate to the **Installed** tab in the Marketplace and select the **Uninstall** button, or from a terminal type `docker extension uninstall IMAGE[:TAG]`.
```
